Source/WebCore/platform/graphics/coretext/FontCoreText.cpp

```diff
@@ -939,4 +939,15 @@ bool Font::hasAnyComplexColorFormatGlyphs(const GlyphBufferGlyph* glyphs, unsign
     return false;
 }
 
+#if USE(APPLE_INTERNAL_SDK)
+#include <WebKitAdditions/FontCoreTextAdditions.cpp>
+#else
+#if ENABLE(MULTI_REPRESENTATION_HEIC)
+FontMetrics Font::metricsForMultiRepresentationHEIC() const
+{
+    return { };
+}
```

Comment:
I do not think returning `FontMetrics` is the right type for the purpose of `MultiRepresentationHEIC`. `FontMetrics` contains the metrics for the whole font. So it contains things like `m_lineGap`, `m_lineSpacing` which are related to fonts not the glyphs.

Surprisingly I could not find a type for glyph metrics. I would suggest adding one for your purpose like 

``` 
struct MultiRepresentationHEICMetrics {
    float width;
    float ascent;
    float descent;
    FloatSize size() const { return ... };
};
```